AppCheck/Sources/Public/AppCheck/GACAppCheck.h

```diff
@@ -35,21 +37,16 @@ FOUNDATION_EXPORT NSString *const kGACAppCheckTokenNotificationKey NS_SWIFT_NAME
 FOUNDATION_EXPORT NSString *const kGACAppCheckAppNameNotificationKey NS_SWIFT_NAME(InternalAppCheckAppNameNotificationKey);
 
 NS_SWIFT_NAME(InternalAppCheck)
-@interface GACAppCheck : NSObject
+@interface GACAppCheck : NSObject <GACAppCheckInterop>
 
 - (instancetype)init NS_UNAVAILABLE;
 
-/// Returns a default instance of `AppCheck`.
-/// @return An instance of `AppCheck` for `FirebaseApp.defaultApp()`.
-/// @throw Throws an exception if the default app is not configured yet or required `FirebaseApp`
-/// options are missing.
-+ (instancetype)appCheck NS_SWIFT_NAME(appCheck());
-
 /// Returns an instance of `AppCheck` for an application.
-/// @param firebaseApp A configured `FirebaseApp` instance if exists.
-/// @return An instance of `AppCheck` corresponding to the passed application.
-/// @throw Throws an exception if required `FirebaseApp` options are missing.
-+ (nullable instancetype)appCheckWithApp:(FIRApp *)firebaseApp NS_SWIFT_NAME(appCheck(app:));
+/// @param app A configured `FirebaseApp` instance.
+/// @param appCheckProvider  An `InternalAppCheckProvider` instance that provides App Check tokens.
```

Comment:
should `InternalAppCheckProvider` be `GACAppCheckProvider`?

Reply:
I was using the Swift name from [`GACAppCheckProvider.h`](https://github.com/firebase/firebase-ios-sdk/blob/a096f4daefb3666dd64f21b7df670d56816571ee/AppCheck/Sources/Public/AppCheck/GACAppCheckProvider.h#L25), should I be using the ObjC name instead?

Note: The name is odd because the FirebaseAppCheck version was already named [`AppCheckProvider`](https://github.com/firebase/firebase-ios-sdk/blob/a096f4daefb3666dd64f21b7df670d56816571ee/FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheckProvider.h#L25). I plan to rename the Swift symbols to `AppCheckCore*`, instead of `InternalAppCheck*`, in another PR.

Reply:
That all SGTM! I think the Swift name is good.

Reply:
No changes needed!